qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -148,7 +148,7 @@ def _find_forward_candidates(self, node_id_t):
 
         if self.template_dag_dep.direct_successors(node_id_t):
             maximal_index = self.template_dag_dep.direct_successors(node_id_t)[-1]
-            for elem in pred:
+            for elem in matches:
                 if elem > maximal_index:
                     pred.remove(elem)
```

Comment:
It's not clear to me that this is correct, because in the lines just out of sight of this diff, `pred` is modified such that it's no longer the same as `matches`.  This change could cause an `IndexError` to be emitted if `node_id_t` ends up greater than `maximal_index` (though I don't know if that's possible).

Reply:
You are correct. I read the lines above, but somehow forgot that `remove` will throw an error if the value is not present.